modules/pallets/messaging-fees/src/impls.rs

```diff
@@ -117,8 +117,7 @@ where
 				else {
 					return None;
 				};
-				let scaling_power = 18u8.saturating_sub(decimals); // assumption that the decimals will always be less than 18
-				let scaling_factor = U256::from(10u128.pow(scaling_power as u32));
+				let scaling_factor = U256::from(10u128.pow(decimals as u32));
```

Comment:
The per_byte_fee is already in the correct decimals for the source chain, don't even think we need this